plugins/libplugin.c

```diff
@@ -619,21 +649,23 @@ struct plugin_timer *plugin_timer(struct plugin_conn *rpc, struct timerel t,
 
 static void plugin_logv(enum log_level l, const char *fmt, va_list ap)
 {
-	char *message;
-
-	printf_json(STDOUT_FILENO,
-		    "{ 'jsonrpc': '2.0', "
-		    "'method': 'log', "
-		    "'params': { 'level': '%s', 'message': \"",
-		    l == LOG_DBG ? "debug"
-		    : l == LOG_INFORM ? "info"
-		    : l == LOG_UNUSUAL ? "warn"
-		    : "error");
-
-	message = tal_vfmt(NULL, fmt, ap);
-	write_all(STDOUT_FILENO, message, strlen(message));
-	printf_json(STDOUT_FILENO, "\" } }\n\n");
-	tal_free(message);
+	struct json_out *jout = json_out_new(tmpctx);
+
+	json_out_start(jout, NULL, '{');
+	json_out_addstr(jout, "jsonrpc", "2.0");
+	json_out_addstr(jout, "method", "log");
+
+	json_out_start(jout, "params", '{');
+	json_out_addstr(jout, "level",
+			l == LOG_DBG ? "debug"
+			: l == LOG_INFORM ? "info"
+			: l == LOG_UNUSUAL ? "warn"
+			: "error");
+	json_out_addv(jout, "message", true, fmt, ap);
+	json_out_end(jout, '}');
+
+	/* Last '}' is done by finish_and_send_json */
+	finish_and_send_json(STDOUT_FILENO, jout);
```

Comment:
~Still strange that we don't use the higher level helpers for `json_stream` here, they'd also check the JSON stream is well formed.~

Nevermind, it migrates to `json_out` in later commits anyway :-)